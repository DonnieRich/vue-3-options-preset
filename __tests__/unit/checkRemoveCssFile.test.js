const { vol } = require("memfs");
jest.mock("fs/promises");

const { removeCssFile } = require("../../src/bin/removeCssFile");

// import all json config files
const { config } = require('../../src/config/config.test');
const { BASE_DIR } = config.get();

describe(removeCssFile, () => {
    beforeEach(() => {
        vol.reset();
    });

    it("Should pass if the file style.css has been found and removed", async () => {

        vol.fromJSON(
            {
                [`${BASE_DIR}/style.css`]: "my style"
            },
            "/"
        );

        await removeCssFile(BASE_DIR);
        expect(vol.toJSON()).toMatchSnapshot();
    });

    it("Should pass if the file main.js has not been removed", async () => {

        const json = {
            [`${BASE_DIR}/main.js`]: "my js"
        };
        vol.fromJSON(
            json,
            "/"
        );

        await removeCssFile(BASE_DIR);
        expect(vol.toJSON()).toEqual(json);
    });
});