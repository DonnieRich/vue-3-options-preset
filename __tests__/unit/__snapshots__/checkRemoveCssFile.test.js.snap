// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`removeCssFile Should pass if the file style.css has been found and removed 1`] = `
{
  "/src": null,
}
`;
